src/main/java/frc/robot/commands/Shoot.java

```diff
@@ -66,19 +96,38 @@ public void initialize() {
                 // Reset the timer
                 shootTimer.reset();
                 shootTimer.stop();
-                shooter.resetPID();
-                drive.setIsAlign(true); // Enable alignment mode on the drivetrain
-                drive.onlyUseTags(new int[]{3, 4, 7, 8});
+
+                // Enable alignment mode on the drivetrain
+                drive.setIsAlign(true);
+
+                // only use the April Tags on the speakers
+                // April Tags 3, 4, 7, and 8 are the speaker tags on the Alliance Wall.
+                // this is a backdoor to vision...
+                drive.onlyUseTags(aprilTagsSpeaker);
+
+                // we are not shooting yet
                 shooting = false;
+
+                // the alliance determines which speaker we target
+                // the speakerPose need not be calculated every execute; do it in initialize()
+                speakerPose = (Robot.getAlliance() == Alliance.Red) ?
+                                VisionConstants.RED_SPEAKER_POSE : VisionConstants.BLUE_SPEAKER_POSE;
         }
+
         @Override
         public void execute() {
+                // TODO: positive x displacement -> left of speaker only for Blue Alliance
                 // Positive x displacement means we are to the left of the speaker
+                // TODO: doesn't positive y displacement mean we are above the speaker?
                 // Positive y displacement means we are below the speaker.
-                Pose3d speakerPose = Robot.getAlliance() == Alliance.Red ?
-                        VisionConstants.RED_SPEAKER_POSE : VisionConstants.BLUE_SPEAKER_POSE;
                 // shooterHeight and shooterOffset have an additional offset because the shooter is offset from the arm, right?
+                // get the direction the robot is facing
                 Rotation2d driveYaw = drive.getYaw();
+
+                // calculate the displacement to the speaker
+                //   actually, this calculates the displacement **from** the speaker to the robot, so Z value is negative...
+                // TODO: 22.7 inches - speaker.z should be negative? speakerPose.getZ() is 80.5 inches.
+                // TODO: these magic numbers should not be here! They should be in the Arm and Shooter subsystems
```

Comment:
You can move them, but we didn't have a lot of time to make this change, so we put them here temporarily.